### ISSUE TITLE
How do I avoid nan gradients for exponential parameters?

### ISSUE THREAD

**Issue by Emmanuel357** [pytorch/pytorch#60338]
Not immediately but as soon as my loss approaches zero when training my neural network with a custom exponential parameter I start getting nan gradients. I don't want to stop using the exponential parameter. Is there a way that I can avoid nan gradients for my exponential parameter or how do I detect that pytorch is about to give my parameter a nan gradient so I can disable the parameter.

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7

**Comment by simaki** [comment]
[detect_anomaly](https://pytorch.org/docs/stable/autograd.html?highlight=detect_anomaly#torch.autograd.detect_anomaly) may help but anyway I think we need a minimal reproducible example to look into that. Also, questions are supposed to be posted to https://discuss.pytorch.org/ as it's more likely you will get help there.

**Comment by albanD** [comment]
Closing as this is not a bug/feature request. We can continue the discussion on the forum if you have follow up.
